components/Map.js

```diff
@@ -0,0 +1,44 @@
+import { useState, useEffect } from "react";
+import ReactMapGL, { Marker } from "react-map-gl";
+export default function Map({ locations }) {
+  // UseState hook
+  const [selectedLocation, setSelectedLocation] = useState({});
+  const [viewport, setViewport] = useState({
+    width: "100%",
+    height: "100%",
+    // The latitude and longitude of the center of London
+    latitude: 40.0718,
+    longitude: -2.13401,
+    zoom: 12.5,
+  });
+  return (
+    <ReactMapGL
+      mapStyle="mapbox://styles/mapbox/streets-v11"
+      mapboxApiAccessToken="pk.eyJ1IjoiaGVjdG9ycGxpbmlvIiwiYSI6ImNrejQ3cDc2ZTBjbHEyb3J4MzMzZHpmMWMifQ.XT3g3xJFTaNGxYeMBtBoaQ"
+      {...viewport}
+      onViewportChange={(nextViewport) => setViewport(nextViewport)}
```

Comment:
I think is not necessary to create a function to send the same parms, you can do 

      onViewportChange={setViewport}

Reply:
yes, its works perfect